### PR TITLE
FIX: Compilation with Windows/Visual Studio 2019

### LINKED ISSUE #104: Git: Cloning problem with case non-sensitive filesystems
While cloning on a typical (sic) MacOS X partition:
warning: the following paths have collided (e.g. case-sensitive paths
on a case-insensitive filesystem) and only one from the same
colliding group is in the working tree:

  'bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_Controller_doc.h'
  'bindings/Sofa/src/SofaPython3/Sofa/Core/binding_controller_doc.h'

and some updates in this file are not applied.
(By default APFS/HFS filesystems are case non-sensitive)
I guess it will be the case with Windows as well.

I don't know if there is a magic trick to fix this problem; I guess the nuclear option to modify the git history is a bit too invasive :)

### PR DESCRIPTION
**Pull request by fredroy**
Before merging, needs Sofa#1162 (and Sofa#1101) to fully compile with Win/VS2019.

Also, fix #104

### CODE REVIEW

**Comment by marques-bruno** [comment]
Thanks for your contribution!
Could one of the admins verify this patch?

File: bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_ForceField.h [hunk 1]
```diff
@@ -39,7 +39,7 @@ namespace sofapython3
 {
 
 template<class TDOFType>
-class SOFAPYTHON3_API ForceField_Trampoline  : public sofa::core::behavior::ForceField<TDOFType>, public PythonTrampoline
+class ForceField_Trampoline  : public sofa::core::behavior::ForceField<TDOFType>, public PythonTrampoline
```
Comment (damienmarchal) [inline bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_ForceField.h]: Removed SOFAPYTHON3_API ?

**Comment by damienmarchal** [comment]
Thank you Fred, can you just explain why removing the "classical" macro *API, is this something new in VS2019 ?

**Comment by fredroy** [comment]
Frankly speaking, for this class, it was a mystery.
I dont know why but the compiler did not let me set the macro.
As it was working without it, well I let it like that... 👋 
Magical !

**Comment by marques-bruno** [comment]
> Frankly speaking, for this class, it was a mystery.
> I dont know why but the compiler did not let me set the macro.
> As it was working without it, well I let it like that... 
> Magical !

I think it's because of the attribute=hidden flag that is hidden in the SOFAPYTHON3_API macro. I think it was a mistake to have it there. I thought that it would be simpler to set the PYBIND11_EXPORT macro in the SOFAPYTHON3_API macro, to simplify writing but it seems to be an over simplification of the cases where the flag should be applied.

In a nutshell, the SOFAPYTHON3_API macro should stay where it is but the PYBIND11_EXPORT macro  should be set individually to each method/class containing pybind symbols in the signature, instead of having it in the API macro.
(must read: [](https://pybind11.readthedocs.io/en/stable/advanced/misc.html))

**Comment by marques-bruno** [comment]
On Linux at least, the removal of the macro on the ForceField does not seem to cause any problems. @fredroy , on windows, can you successfully run the python tests related to forcefields & controllers? (in bindings/Sofa/tests/Core/[....]ForceField.py for instance)

**Comment by marques-bruno** [comment]
[ci-build]

**Pull request merged**